wideq/ac.py

```diff
@@ -134,49 +258,232 @@ def set_fan_speed(self, speed):
-
+        
         mode_value = self.model.enum_value('OpMode', mode.value)
         self._set_control('OpMode', mode_value)
 
     def set_on(self, is_on):
         """Turn on or off the device (according to a boolean).
         """
 
-        op = ACOp.RIGHT_ON if is_on else ACOp.OFF
+        op = ACOp.ALL_ON if is_on else ACOp.OFF
         op_value = self.model.enum_value('Operation', op.value)
         self._set_control('Operation', op_value)
 
+    def set_wind_leftright(self, mode):
+        """Set the device's wind direction mode to an `WDirLeftRight` value.
+        """
+        
+        wdir_value = self.model.enum_value('WDirLeftRight', mode.value)
+        self._set_control('WDirLeftRight', wdir_value)
+
+    def set_wdirhstep(self, mode):
+        """Set the device's wind horizontal direction mode to an `WDirHStep` value.
+        """
+
+        self._set_control_ac_wdirstep('WDirHStep',int(mode.value))
+
+    def set_wdirvstep(self, mode):
+        """Set the device's wind vertical direction mode to an `WDirVStep` value.
+        """
+
+        self._set_control_ac_wdirstep('WDirVStep',int(mode.value))
+
+    def set_airclean(self, is_on):
+        """Set the device's air purifier mode to an `AirClean` value.
+        """
+
+        mode = AIRCLEAN.ON if is_on else AIRCLEAN.OFF
+        mode_value = self.model.enum_value('AirClean', mode.value)
+        self._set_control('AirClean', mode_value)
+
+    def set_etc_mode(self, name, is_on):
+        """Set the device's extra function mode to each value.
+        """
+
+        mode = ACETCMODE.ON if is_on else ACETCMODE.OFF
+        mode_value = self.model.enum_value(name, mode.value)
+        self._set_control(name, mode_value)
+
+    def set_sleep_time(self, sleeptime):
+       """Set the device's sleap timeer to an `SleepTime` value.
+        """
+        self._set_control('SleepTime', sleeptime)
+
     def get_filter_state(self):
         """Get information about the filter."""
-
+        
         return self._get_config('Filter')
-
+    
     def get_mfilter_state(self):
         """Get information about the "MFilter" (not sure what this is).
         """
 
         return self._get_config('MFilter')
-
+    
     def get_energy_target(self):
         """Get the configured energy target data."""
-
+        
         return self._get_config('EnergyDesiredValue')
-
+    
     def get_light(self):
         """Get a Boolean indicating whether the display light is on."""
-
+        
         value = self._get_control('DisplayControl')
         return value == '0'  # Seems backwards, but isn't.
-
+    
     def get_volume(self):
         """Get the speaker volume level."""
-
+        
         value = self._get_control('SpkVolume')
         return int(value)
 
-    def poll(self):
-        """Poll the device's current state.
+    def get_outdoor_weather(self, area):
+        """Get outdoor weather"""
+
+        data = self.client.session.get_outdoor_weather(area)
+        return data
```

Comment:
I don't think this is necessary here—the API is associated with the session, not the device.